aloha-core/src/main/scala/com/eharmony/aloha/models/exploration/EpsilonGreedyModel.scala

```diff
@@ -0,0 +1,116 @@
+package com.eharmony.aloha.models.exploration
+
+import com.eharmony.aloha.factory.{ModelFactory, ModelParser, ParserProviderCompanion}
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.models.{BaseModel, Model}
+import com.eharmony.aloha.score.Scores.Score
+import com.eharmony.aloha.score.basic.ModelOutput
+import com.eharmony.aloha.score.conversions.ScoreConverter
+import com.eharmony.aloha.score.conversions.ScoreConverter.Implicits.IntScoreConverter
+import com.eharmony.aloha.semantics.Semantics
+import com.eharmony.aloha.semantics.func.GenAggFunc
+import com.mwt.explorers.EpsilonGreedyExplorer
+import com.mwt.policies.Policy
+
+import scala.collection.{immutable => sci}
+
+/**
+  * Since explore-java has chosen to force the Policy to evaluate we can't just evaluate the model inside of the policy.
+  * Instead the Policy acts as a pass through for the model which is evaluated externally.  This inversion of control
+  * allows the model to fail before evaluating the Policy.
+  */
+private[this] case object ModelPolicy extends Policy[Int] {
+  override def chooseAction(action: Int): Int = action
+}
+
+/**
+  * A model which does epsilon greedy style exploration.  This will choose a random action with probability epsilon
+  * or an action from the defaultPolicy with probability 1 - epsilon.
+  * @param modelId a model identifier
+  * @param defaultPolicy the model to use for exploitation.  This MUST be deterministic for the probability to be correct.
+  *                      The model must return a value in the range 1 to `classLabels.size` (inclusive).
+  * @param epsilon the exploration/exploitation tradeoff parameter
```

Comment:
epsilon must be in the interval [0, 1]. 0 indicates never select an action randomly, 1 always select an action randomly.